File: Selenium/Activities/Activity7.py
from selenium import webdriver
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.by import By

service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    driver.get("https://training-support.net/selenium/drag-drop")
    print("The title is: ", driver.title)
    #Find the ball and simulate a click and drag to move it into "Dropzone 1".
    #Verify that the ball has entered Dropzone 1.
    #Once verified, move the ball into "Dropzone 2".
    #Verify that the ball has entered Dropzone 2.
    actions = ActionChains(driver)
    ball = driver.find_element(By.ID, "draggable")
    dropzone1 = driver.find_element(By.ID, "droppable")
    dropzone2 = driver.find_element(By.ID, "dropzone2")
    actions.drag_and_drop(ball, dropzone1).perform()
    if(dropzone1.text.find("Dropped!")):
        print("The ball is in Dropzone 1")
    actions.drag_and_drop(dropzone1, dropzone2).perform()
    if(dropzone2.text.find("Dropped!")):
        print("The ball is in Dropzone 2")

File: Selenium/Activities/activity7.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class activity7 {
    public static void main(String[] args) {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/selenium/drag-drop");
        System.out.println("The title is: " +driver.getTitle());
        //Find the ball and simulate a click and drag to move it into "Dropzone 1".
        Actions builder = new Actions(driver);
        WebElement ball = driver.findElement(By.id("draggable"));
        WebElement dropZone1 = driver.findElement(By.id("droppable"));
        builder.click(ball).dragAndDrop(ball,dropZone1).build().perform();
        //Verify that the ball has entered Dropzone 1.
        driver.findElement(By.cssSelector("#draggable")).getCssValue("Color");
        System.out.println("The ball is in Dropzone1");
        //Once verified, move the ball into "Dropzone 2".
        WebElement dropZone2 = driver.findElement(By.id("dropzone2"));
        builder.dragAndDrop(dropZone1, dropZone2).build().perform();
        //Verify that the ball has entered Dropzone 2.
        String check2 = driver.findElement(By.cssSelector("#dropzone2")).getText();
        System.out.println("the text in dropzone2 is: " + check2);
        String text = "Dropzone 2\n" +
                "Dropped!";
        boolean value = check2.contains(text);
        if (value){
            System.out.println("The ball is moved to Dropzone2");}
        driver.quit();
    }
}
